### PR TITLE
User can create an order by adding the first product

### PR DESCRIPTION
**Pull request by miyeaier**
## User Story

```
As a user
In order to start buying product
I would like an order to be created when I add my first product
```
Screenshot 2022-05-06 at 13 22 37
Screenshot 2022-05-06 at 13 22 09

### CODE REVIEW

File: src/components/Products.jsx [hunk 1]
```diff
@@ -14,6 +15,22 @@ const Products = () => {
     fetchProducts();
   }, []);
 
+  const addToOrder = async (id) => {
+    const toastSetting = { autoClose: 1500, toastId: "message-box" };
+    if (orderInProgress === false) {
+      const response = await axios.post("https://reqres.in/api/orders", {
```
Comment (tochman) [inline src/components/Products.jsx]: Why are you using Reqres?

File: src/components/Products.jsx [hunk 2]
```diff
@@ -14,6 +15,22 @@ const Products = () => {
...
+        params: { product_id: id },
+      });
+      setOrderInProgress(true);
+      toast(response.data.message, toastSetting);
+    } else {
+      const response = await axios.put("https://reqres.in/api/orders", {
```
Comment (tochman) [inline src/components/Products.jsx]: This is out of scope. the PUT request is used for updating the order. The question is if you are in control of what this code does?

File: src/components/Products.jsx [hunk 3]
```diff
@@ -26,12 +43,29 @@ const Products = () => {
...
+});
+  return(
+  <>
+  <Container>
+    <List inverted id="products-list" size="big">
+      {productlist}
+    </List>
+  </Container>
+  <ToastContainer data-cy="message-box" />
+</>
```
Comment (tochman) [inline src/components/Products.jsx]: Please get into the habit of formatting your code.

**Review by tochman — COMMENTED** [review]
there is a scope issue in my opinion.

File: src/components/Products.jsx (hunk 2 above)
Comment (miyeaier) [reply src/components/Products.jsx]: > This is out of scope. the PUT request is used for updating the order. The question is if you are in control of what this code does?

yes. I don't really understand this part 🫣. but I will learn and modify this today ( I just cope with old code and if no this part it will not work so even I am Confuse but feel works so I used it, and I know that's not right🫠🫢 )

File: src/components/Products.jsx (hunk 1 above)
Comment (miyeaier) [reply src/components/Products.jsx]: > Why are you using Reqres?

I will correct this today

File: src/components/Products.jsx (hunk 3 above)
Comment (miyeaier) [reply src/components/Products.jsx]: no problem. thank you for mention!

File: src/components/Products.jsx [hunk 4]
```diff
@@ -13,6 +14,13 @@ const Products = () => {
   useEffect(() => {
     fetchProducts();
   }, []);
+  const addToOrder = async (id) => {
+    const response = await axios.post("https://reqres.in/api/orders", {
```
Comment (maxarvid) [inline src/components/Products.jsx]: Looks like you are still using reqres here

File: src/components/Products.jsx [hunk 5]
```diff
@@ -13,6 +14,13 @@ const Products = () => {
...
+      params: { product_id: id },
+    });
+    toast(response.data.message, { toastId: "message-box" });
```
Comment (maxarvid) [inline src/components/Products.jsx:22]: The toast does not run as intended on my machine. Does it really run on yours?

Skärmavbild 2022-05-06 kl  10 54 19

File: src/components/Products.jsx [hunk 6]
```diff
@@ -26,12 +34,28 @@ const Products = () => {
...
-
-  return <div data-cy="product-list">{productlist}</div>;
+  return (
+    <>
+      <Container>
+        <List inverted id="products-list" size="big">
+          {productlist}
+        </List>
+      </Container>
+      <ToastContainer data-cy="message-box" />
```
Comment (maxarvid) [inline src/components/Products.jsx]: Here you are giving the `ToastContainer` a `data-cy`, but in the test you are looking for its `id`. Pick one or the other.

File: src/components/Products.jsx [hunk 7]
```diff
@@ -26,12 +34,28 @@ const Products = () => {
           <Item.Header>{product.name}</Item.Header>
           <Item.Meta> {`${product.price}kr`}</Item.Meta>
           <Item.Description>{product.description}</Item.Description>
+          <button
+            data-cy="order-button"
+            onClick={() => addToOrder(product.id)}
```
Comment (maxarvid) [inline src/components/Products.jsx:40]: Does the `onClick` event need to be a callback here?

**Review by maxarvid — CHANGES_REQUESTED** [review]
Moving forward! I made a few comments and suggestions.

File: src/components/Products.jsx (hunk 5 above)
Comment (miyeaier) [reply src/components/Products.jsx:22]: no 🫣

File: src/components/Products.jsx (hunk 4 above)
Comment (miyeaier) [reply src/components/Products.jsx]: > Looks like you are still using reqres here

I will fix soon 🫠

File: src/components/Products.jsx (hunk 6 above)
Comment (miyeaier) [reply src/components/Products.jsx]: > Here you are giving the `ToastContainer` a `data-cy`, but in the test you are looking for its `id`. Pick one or the other.

good suggestion i will change now

File: src/components/Products.jsx (hunk 2 above)
Comment (maxarvid) [reply src/components/Products.jsx]: We've discussed the difference.

File: src/components/Products.jsx (hunk 7 above)
Comment (maxarvid) [reply src/components/Products.jsx:40]: For curious readers: We experimented together, and yes, the `() =>` is necessary.

**Review by maxarvid — APPROVED** [review]
LGTM
